### PR TITLE
solution

### PR DESCRIPTION
**Pull request by maksym-kostetskyi**

### CODE REVIEW

File: src/averageAges.js [hunk 1]
```diff
@@ -1,61 +1,42 @@
...
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenSorted = (withChildren !== undefined)
+    ? people.filter(
+      person => person.sex === 'f' && people.some(
+        x => x.mother === person.name))
+    : people.filter(person => person.sex === 'f');
+
+  return womenSorted.reduce(
+    (sum, x) => sum + x.died - x.born, 0) / womenSorted.length;
```
Comment (loralevitska) [inline src/averageAges.js:21]: Combine repeatable code with helper function and reuse it

File: src/averageAges.js [hunk 2]
```diff
@@ -1,61 +1,42 @@
...
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const withMotherSorted = (onlyWithSon !== undefined)
+    ? people.filter(
+      person => person.sex === 'm' && people.some(
+        x => x.name === person.mother))
+    : people.filter(person => people.some(
+      x => x.name === person.mother));
```
Comment (loralevitska) [inline src/averageAges.js:30]: same here: replace repeatable code with helper function and reuse it. Correct all repetitions throughout your solution

**Review by loralevitska — CHANGES_REQUESTED** [review]
Good job! There are things that should be fixed:

File: src/averageAges.js [hunk 3]
```diff
@@ -1,61 +1,54 @@
...
+  return people.filter(person => person.mother === x.name);
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const menSorted = (century !== undefined)
```
[the remark this answers is not in the document]
Comment (DarkMistyRoom) [reply src/averageAges.js:25]: fix it everywhere

**Review by DarkMistyRoom — CHANGES_REQUESTED** [review]
Great job. Let's try to simplify it a bit

**Review by etojeDenys — APPROVED** [review]
good job

**Pull request closed without merging**